mne/preprocessing/maxwell.py

```diff
@@ -41,6 +41,114 @@
+        comparison to the experimental recording. If a list, you can explicitly
+        provide the list of channel names to be marked as bad.
+
+        .. note::
+           Non-MEG channels are silently dropped from the list of bads.
+    %(verbose)s
+
+    Returns
+    -------
+    raw_er_prepared : instance of Raw
+        A copy of the passed empty-room recording, ready for Maxwell filtering.
+
+    Notes
+    -----
+    This function will:
+
+    * Compile the list of bad channels according to the ``bads`` parameter.
+    * Inject the device-to-head transformation matrix from the experimental
+      recording into the empty-room recording.
+    * Set the following properties of the empty-room recording to match the
+      experimental recording:
+
+      * Montage
+      * Measurement date
+      * Annotations
+
+    If no modification of the measurement date or annotations is desired,
+    we suggest saving them and reverting them afterward, e.g. with::
+
+        >>> orig_annot, meas_date = raw_er.annotations, raw_er.info['meas_date']  # doctest:+SKIP
+        >>> raw_er_prepared = prepare_empty_room(raw_er, raw=raw)  # doctest:+SKIP
+        >>> raw_er_prepared.set_meas_date(meas_date)  # doctest:+SKIP
+        >>> annot.onset += (raw_er_prepared.first_samp - raw_er.first_samp) / raw_er.info['sfreq']  # doctest:+SKIP
+        >>> raw_er_prepared.set_annotations(orig_annot)  # doctest:+SKIP
+
+    Note that setting the ``meas_date`` back is necessary in order to revert
+    the annotations, because the onsets are relative to the ``meas_date`` (plus
+    first samp, which is why it is adjusted to reflect the new first_samp).
+
+    .. versionadded:: 1.1
+    """  # noqa: E501
+    _validate_type(item=raw_er, types=BaseRaw, item_name='raw_er')
+    _validate_type(item=raw, types=BaseRaw, item_name='raw')
+    _validate_type(item=bads, types=(list, str), item_name='bads')
+    if isinstance(bads, str):
+        _check_option(
+            parameter='bads', value=bads, allowed_values=['from_raw', 'union']
+        )
+
+    raw_er_prepared = raw_er.copy()
+    del raw_er  # just to be sure
+
+    # handle bads; only keep MEG channels
+
+    if bads == 'from_raw':
+        bads = raw.info['bads']
+    elif bads == 'union':
+        bads = sorted(
+            set(raw.info['bads'] + raw_er_prepared.info['bads'])
+        )
+    bads = [ch_name for ch_name in bads
+            if ch_name.startswith('MEG')]
+    raw_er_prepared.info['bads'] = bads
+
+    # handle dev_head_t
+    raw_er_prepared.info['dev_head_t'] = raw.info['dev_head_t']
+
+    # handle montage
+    montage = raw.get_montage()
+    raw_er_prepared.set_montage(montage)
+
+    # handle first_samp
+    raw_er_prepared._cropped_samp = raw._cropped_samp
+
+    # handle meas_date
+    raw_er_prepared.set_meas_date(raw.info['meas_date'])
+
+    # handle annotations
+    raw_er_prepared.set_annotations(raw.annotations)
```

Comment:
this one is maybe a bit dangerous no? imagine ER file already contains
relevant annotation eg of really bad segment or acq skip or something?
can you maybe add a comment for why each step is necessary? maybe we
need to do this only if we want movement compensation?

Reply:
> this one is maybe a bit dangerous no?

I could imagine use cases where you don't want it. But I definitely have use cases where you do (e.g., movecomp). The motivation is in the test, e.g., with `annotate_movement` if you've already `set_annotations` with it for `raw` you want to carry it over.

It's not clear to me what the best behavior is here. We might want switchable kwargs, like `first_samp : bool`, `annotations : bool`, and `meas_date : bool`. Not sure what the safest defaults are here...

Reply:
We could alternatively offer to merge the existing annotations (if any) with those from raw. We already allow for a union of bads, why not do the same for annotations too?

Reply:
Agreed, looks like we could add `annotations : 'from_raw' | 'union' | 'keep'`, `meas_date : 'from_raw' | 'keep'`, and `first_samp : 'from_raw' | 'keep'`.